app/controllers/root_controller.rb

```diff
@@ -273,10 +259,47 @@ def treat_as_standard_html_request?
     !request.format.json? and !request.format.print? and !request.format.video?
   end
 
-  def local_transaction_details(artefact, authority_slug, snac)
-    raise RecordNotFound if snac.blank? && authority_slug.present?
+  def local_transaction_details(artefact, authority_slug)
+    if authority_slug
+      lgsl = artefact['details']['lgsl_code']
+      lgil = artefact['details']['lgil_override']
 
-    artefact['details'].slice('local_authority', 'local_service', 'local_interaction')
+      local_links_manager_response = local_links_manager_request(authority_slug, lgsl, lgil)
+      artefact['details'].slice('local_service').merge(local_links_manager_response)
+    else
+      artefact['details'].slice('local_service')
+    end
+  end
+
+  def local_links_manager_request(authority_slug, lgsl, lgil)
+    # RESTful endpoint
+    # url = "/local_authorities/#{authority_slug}/services/#{lgsl}/interactions/#{lgil}.json"
+
+    # Non-RESTful endpoint
+    url = "/api/local_interaction_details.json?authority_slug=#{authority_slug}&lgsl=#{lgsl}&lgil=#{lgil}"
+
+    # There will be a wrapper in gds-api-adapters for LocalLinksManager
+    # LocalLinksManager.get(url)
+
+    # this is the hash that was previously added to the artefact['details'] after
+    # the artfact was updated with local_transaction details by calling
+    # `fetch_artefact`
+    # Instead we return this hash from LocalLinksManager's API endpoint
+
+    # For demonstration purposes we return this stubbed response
+    {
+      "local_authority"=>{
+        "name"=>"London Borough of Camden",
+        # "snac"=>"00AG",
+        # "tier"=>"unitary",
+        "homepage_url"=>"http://www.camden.gov.uk/"
+      },
+      "local_interaction"=>{
+        # "lgsl_code"=>432,
+        # "lgil_code"=>0,
+        "url"=>"https://www.westminster.gov.uk/stray-and-lost-dogs"
+      }
```

Comment:
So we're returning 2 different sets of data of here, one for the local authority and one for the local interaction. Wondering whether this should then be 2 different requests. One that hits a LocalLinksManager `/api/local_authorities` endpoint and one that hits `api/local_interaction_details`. So we would make 2 requests and still merge the data in here. But maybe that would be a cleaner separation of the 2 types of data. As a bonus, this could give us a API endpoint in the LocalLinksManager API for later 'Find my local authority'?